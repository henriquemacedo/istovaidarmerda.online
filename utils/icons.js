const icons = {
  twitter:
    "M24.091 11.698c0.013 0.186 0.013 0.373 0.013 0.557 0 5.665-4.311 12.191-12.191 12.191-0.009 0-0.020 0-0.031 0-2.433 0-4.698-0.717-6.595-1.952l0.047 0.029c0.292 0.034 0.63 0.054 0.973 0.054 0.022 0 0.043-0 0.065-0l-0.003 0c0.007 0 0.014 0 0.022 0 2.007 0 3.854-0.689 5.315-1.844l-0.018 0.014c-1.868-0.035-3.441-1.257-3.999-2.941l-0.009-0.030c0.242 0.040 0.522 0.064 0.807 0.066h0.002c0.401-0.001 0.79-0.055 1.159-0.153l-0.032 0.007c-1.974-0.41-3.436-2.134-3.436-4.2 0-0.002 0-0.004 0-0.005v0-0.053c0.558 0.317 1.221 0.516 1.928 0.544l0.008 0c-1.158-0.781-1.909-2.087-1.909-3.569 0-0.796 0.217-1.542 0.595-2.181l-0.011 0.020c2.123 2.598 5.263 4.297 8.804 4.482l0.030 0.001c-0.066-0.294-0.105-0.633-0.107-0.98v-0.002c0.003-2.367 1.922-4.284 4.289-4.284 1.231 0 2.34 0.518 3.122 1.348l0.002 0.002c1.025-0.201 1.937-0.561 2.758-1.056l-0.038 0.021c-0.332 1.007-0.998 1.828-1.864 2.35l-0.019 0.011c0.915-0.105 1.75-0.338 2.525-0.682l-0.055 0.022c-0.598 0.863-1.309 1.596-2.124 2.198l-0.025 0.018z",
  youtube:
    "M26.232 11.355c-0.245-0.902-0.939-1.6-1.821-1.846l-0.018-0.004c-1.622-0.438-8.126-0.438-8.126-0.438s-6.507 0-8.126 0.437c-0.9 0.251-1.594 0.948-1.835 1.832l-0.004 0.018c-0.435 1.633-0.435 5.039-0.435 5.039s0 3.406 0.435 5.039c0.247 0.895 0.943 1.583 1.823 1.817l0.018 0.004c1.622 0.437 8.126 0.437 8.126 0.437s6.507 0 8.126-0.437c0.898-0.238 1.592-0.927 1.835-1.803l0.004-0.018c0.435-1.633 0.435-5.039 0.435-5.039s-0.002-3.405-0.437-5.038zM14.139 19.486v-6.185l5.436 3.093z",
  "sign-language":
    "M9.082 27.874c-0.016-0.834 0.683-1.517 1.517-1.517h3.244v-0.298h-6.555c-0.834 0-1.533-0.683-1.517-1.517 0.016-0.809 0.676-1.459 1.488-1.459 0 0 0 0 0 0h6.585v-0.298h-8.006c-0.834 0-1.533-0.683-1.517-1.517 0.016-0.809 0.676-1.459 1.488-1.459 0 0 0 0 0 0h8.036v-0.298h-6.519c-0.834 0-1.533-0.683-1.517-1.517 0.016-0.809 0.676-1.459 1.488-1.459 0 0 0 0 0 0h8.779l-1.619-1.178c-0.373-0.274-0.613-0.711-0.613-1.204 0-0.822 0.666-1.488 1.488-1.488 0.329 0 0.633 0.107 0.879 0.288l-0.004-0.003 6.425 4.673c0.448 0.328 0.735 0.853 0.735 1.444v0 7.351c0 0 0 0 0 0 0 0.841-0.582 1.546-1.364 1.736l-0.012 0.003-4.191 0.986c-0.491 0.121-1.056 0.19-1.636 0.19h-5.594c-0 0-0 0-0 0-0.812 0-1.471-0.65-1.488-1.458l-0-0.002zM12.897 16.119h1.897l-0.583-0.424c-0.477-0.35-0.784-0.909-0.784-1.54 0-0.555 0.238-1.055 0.617-1.403l0.001-0.001-0.677-0.87c-0.513-0.658-1.481-0.79-2.129-0.265-0.337 0.275-0.551 0.691-0.551 1.156 0 0.346 0.118 0.665 0.316 0.918l-0.002-0.003 1.894 2.433zM27.652 16.009l-0.261-7.94c-0.027-0.801-0.683-1.439-1.487-1.439-0.822 0-1.488 0.666-1.488 1.488 0 0.017 0 0.034 0.001 0.051l-0-0.002 0.066 2.001-5.394-6.927c-0.275-0.351-0.698-0.574-1.174-0.574-0.336 0-0.646 0.111-0.895 0.299l0.004-0.003c-0.668 0.5-0.778 1.471-0.265 2.129l4.005 5.143-0.235 0.183-4.937-6.34c-0.275-0.351-0.698-0.574-1.174-0.574-0.336 0-0.646 0.111-0.895 0.299l0.004-0.003c-0.668 0.5-0.778 1.471-0.265 2.129l4.919 6.317-0.235 0.183-4.046-5.196c-0.275-0.351-0.698-0.574-1.174-0.574-0.336 0-0.646 0.111-0.895 0.299l0.004-0.003c-0.668 0.5-0.778 1.471-0.265 2.129l2.711 3.482c0.296-0.199 0.66-0.317 1.052-0.317 0.421 0 0.81 0.137 1.125 0.368l-0.005-0.004 6.425 4.673c0.552 0.406 0.906 1.052 0.907 1.781v0.886l3.18-2.477c0.421-0.33 0.689-0.838 0.689-1.409 0-0.021-0-0.041-0.001-0.062l0 0.003z",
};

export default icons;
