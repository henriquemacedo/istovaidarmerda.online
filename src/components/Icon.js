import React from "react";
import PropTypes from "prop-types";

import icons from "../../utils/icons";

const Icon = ({ fill, icon }) => (
	<svg viewBox="0 0 32 32">
		<path style={{ fill }} d={icons[icon]} />
	</svg>
);

Icon.defaultProps = {
	fill: "#ffffff",
};

Icon.propTypes = {
	fill: PropTypes.string,
	icon: PropTypes.string.isRequired,
};

export default Icon;
