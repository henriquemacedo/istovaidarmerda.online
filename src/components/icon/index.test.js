/**
 * The copyright of this file belongs to Feedzai. The file cannot be
 * reproduced in whole or in part, stored in a retrieval system,
 * transmitted in any form, or by any means electronic, mechanical,
 * photocopying, or otherwise, without the prior permission of the owner.
 *
 * © 2020 Feedzai, Strictly Confidential
 */

/**
 * index.test
 *
 * Test suite for the Icon component
 *
 * @author João Dias <joao.dias@feedzai.com>
 * @since ```feedzai.next.release```
 */
import React from "react";
import Logger from "js-logger";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Icon from "./index";
import icons from "../../../utils/icons";

describe("src/components/icon", () => {
  /**
   * @type {import(".").IIconProps}
   */
  let props;

  beforeEach(() => {
    Logger.error = jest.fn();

    props = {
      icon: "twitter",
    };
  });

  afterEach(() => {
    Logger.error.mockRestore();
  });

  describe("render", () => {
    it("should render without errors", () => {
      const component = render(<Icon icon={props.icon} />);

      expect(component).toMatchSnapshot();
    });

    it("should render the default fill color", () => {
      render(<Icon icon={props.icon} />);

      const icon = screen.getByRole("presentation");
      const iconPath = icon.querySelector("path");

      expect(iconPath).toHaveAttribute("fill", "var(--light, #ffffff)");
    });

    it("should render a custom fill color", () => {
      const customProps = {
        ...props,
        fill: "var(--lightBrown, #946037)",
      };

      render(<Icon icon={customProps.icon} fill={customProps.fill} />);

      const icon = screen.getByRole("presentation");
      const iconPath = icon.querySelector("path");

      expect(iconPath).toHaveAttribute("fill", customProps.fill);
    });

    it("should render another icon", () => {
      /**
       * @type {import(".").IIconProps}
       */
      const customProps = {
        ...props,
        icon: "youtube",
      };

      render(<Icon icon={customProps.icon} />);

      const icon = screen.getByRole("presentation");
      const iconPath = icon.querySelector("path");

      expect(iconPath).toHaveAttribute("d", icons.youtube);
    });

    it("should throw an error if the icon is nil", () => {
      /**
       * @type {import(".").IIconProps}
       */
      const customProps = {
        ...props,
        icon: "instagram",
      };

      render(<Icon icon={customProps.icon} />);

      expect(Logger.error).toHaveBeenCalledTimes(1);
      expect(Logger.error).toHaveBeenCalledWith("components/icon.getPathToBeDrawn: ", "Please provide a valid icon");
    });
  });
});
