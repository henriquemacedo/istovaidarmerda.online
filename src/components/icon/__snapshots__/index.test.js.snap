// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/icon render should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        role="presentation"
        viewBox="0 0 32 32"
      >
        <path
          d="M24.091 11.698c0.013 0.186 0.013 0.373 0.013 0.557 0 5.665-4.311 12.191-12.191 12.191-0.009 0-0.020 0-0.031 0-2.433 0-4.698-0.717-6.595-1.952l0.047 0.029c0.292 0.034 0.63 0.054 0.973 0.054 0.022 0 0.043-0 0.065-0l-0.003 0c0.007 0 0.014 0 0.022 0 2.007 0 3.854-0.689 5.315-1.844l-0.018 0.014c-1.868-0.035-3.441-1.257-3.999-2.941l-0.009-0.030c0.242 0.040 0.522 0.064 0.807 0.066h0.002c0.401-0.001 0.79-0.055 1.159-0.153l-0.032 0.007c-1.974-0.41-3.436-2.134-3.436-4.2 0-0.002 0-0.004 0-0.005v0-0.053c0.558 0.317 1.221 0.516 1.928 0.544l0.008 0c-1.158-0.781-1.909-2.087-1.909-3.569 0-0.796 0.217-1.542 0.595-2.181l-0.011 0.020c2.123 2.598 5.263 4.297 8.804 4.482l0.030 0.001c-0.066-0.294-0.105-0.633-0.107-0.98v-0.002c0.003-2.367 1.922-4.284 4.289-4.284 1.231 0 2.34 0.518 3.122 1.348l0.002 0.002c1.025-0.201 1.937-0.561 2.758-1.056l-0.038 0.021c-0.332 1.007-0.998 1.828-1.864 2.35l-0.019 0.011c0.915-0.105 1.75-0.338 2.525-0.682l-0.055 0.022c-0.598 0.863-1.309 1.596-2.124 2.198l-0.025 0.018z"
          fill="var(--light, #ffffff)"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      role="presentation"
      viewBox="0 0 32 32"
    >
      <path
        d="M24.091 11.698c0.013 0.186 0.013 0.373 0.013 0.557 0 5.665-4.311 12.191-12.191 12.191-0.009 0-0.020 0-0.031 0-2.433 0-4.698-0.717-6.595-1.952l0.047 0.029c0.292 0.034 0.63 0.054 0.973 0.054 0.022 0 0.043-0 0.065-0l-0.003 0c0.007 0 0.014 0 0.022 0 2.007 0 3.854-0.689 5.315-1.844l-0.018 0.014c-1.868-0.035-3.441-1.257-3.999-2.941l-0.009-0.030c0.242 0.040 0.522 0.064 0.807 0.066h0.002c0.401-0.001 0.79-0.055 1.159-0.153l-0.032 0.007c-1.974-0.41-3.436-2.134-3.436-4.2 0-0.002 0-0.004 0-0.005v0-0.053c0.558 0.317 1.221 0.516 1.928 0.544l0.008 0c-1.158-0.781-1.909-2.087-1.909-3.569 0-0.796 0.217-1.542 0.595-2.181l-0.011 0.020c2.123 2.598 5.263 4.297 8.804 4.482l0.030 0.001c-0.066-0.294-0.105-0.633-0.107-0.98v-0.002c0.003-2.367 1.922-4.284 4.289-4.284 1.231 0 2.34 0.518 3.122 1.348l0.002 0.002c1.025-0.201 1.937-0.561 2.758-1.056l-0.038 0.021c-0.332 1.007-0.998 1.828-1.864 2.35l-0.019 0.011c0.915-0.105 1.75-0.338 2.525-0.682l-0.055 0.022c-0.598 0.863-1.309 1.596-2.124 2.198l-0.025 0.018z"
        fill="var(--light, #ffffff)"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
